### PR TITLE
Add typing for `_wait_for_object.py`

### PR DESCRIPTION
**Pull request by CoolCat467**
This PR adds type annotations to `_wait_for_object.py` and `_core/_windows_cffi.py`

I would love feedback and any comments on how this could be improved to be more accurate.

### CODE REVIEW

**Review by TeamSpen210 — APPROVED** [review]

**Comment by TeamSpen210** [comment]
I think you might need to access it via `cffi.FFI.CData`, on the class.

**Comment by CoolCat467** [comment]
If anyone has any ideas, the only issue the CI has is the mypy run for windows, saying
```console
trio/_core/_windows_cffi.py:225: error: Name "ffi.CData" is not defined  [name-defined]
```

**Review by jakkdl — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -70,7 +69,6 @@ module = [
 "trio/_core/_generated_io_windows",
```
Comment (jakkdl) [inline pyproject.toml:69]: remove the ignores from pyproject.toml

**Review by jakkdl — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (CoolCat467) [reply pyproject.toml:66]: I think I have gotten requested removed in, 3e9c132, unresolve if more are required.

**Review by jakkdl — APPROVED** [review]

**Pull request merged**